### ISSUE TITLE
events usage incompatible with io.js

### ISSUE THREAD

**Issue by aldafu** [sequelize/sequelize#2933]
io.js is the new Node.js fork by most of the active previous Node.js developers. They've just made their first release available, [changes here](https://github.com/iojs/io.js/blob/545959468fbc878e7f0a486264eebbdf06a57bea/CHANGELOG.md).

[```Promise.prototype.on```](https://github.com/sequelize/sequelize/blob/c3f2603048ac9ffaea0886fd6c135d21bf094428/lib/promise.js#L118) in sequelize calls [```EventEmitter.prototype.addListener```](https://github.com/iojs/io.js/blob/3e1b1dd4a9ac048105a4dc4cd81578e26d39a1fc/lib/events.js#L154) which in turn calls ```this.getMaxListeners```. But ```this``` is now refering to Promise in sequelize, so it doesn't have ```getMaxListeners``` and results in "undefined is not a function" error.

This is the relevant part of a stack trace:
TypeError: undefined is not a function
    at Promise.addListener (events.js:154:18)
    at Promise.on (project\node_modules\sequelize\lib\promise.js:116:31)
    at Promise.<anonymous> (project\node_modules\sequelize\lib\promise.js:261:10)
    at Array.forEach (native)
    at Promise.proxy (project\node_modules\sequelize\lib\promise.js:260:18)
    at Promise.proxySql (project\node_modules\sequelize\lib\promise.js:271:15)
    at Promise._then (project\node_modules\sequelize\lib\promise.js:57:10)
    at Promise$then [as then] (project\node_modules\sequelize\node_modules\bluebird\js\main\promise.js:144:17)
    at Promise.on (project\node_modules\sequelize\lib\promise.js:113:10)

**Comment by mickhansen** [comment]
PR welcome.
We're waiting for Travis support before we can officially support io.js

**Comment by aldafu** [comment]
Seem to only happen when using deprecated functions, e.g. ```on('success'... ``` instead of ```then()``` on sequelize promises.

**Comment by mickhansen** [comment]
@aldafu Yes there are likely still tests using the old pattern, we haven't removed them completely yet although we'd like to.

**Comment by cusspvz** [comment]
related with #2882

**Comment by janmeier** [comment]
Closing this in favor of https://github.com/sequelize/sequelize/issues/2966
